provider/pkg/provider/provider.go

```diff
@@ -1011,6 +1114,14 @@ func (k *kubeProvider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (
 	// is given to it if it's not already provided.
 	//
 
+	urn := resource.URN(req.GetUrn())
+	if isHelmRelease(urn) {
+		if !k.clusterUnreachable {
+			return k.helmReleaseProvider.Check(ctx, req, !k.suppressHelmReleaseBetaWarning)
+		}
+		return nil, fmt.Errorf("can't use Helm Release with unreachable cluster. Reason: %q", k.clusterUnreachableReason)
```

Comment:
Perhaps we should print some sort of placeholder value rather than erroring in this case?

Reply:
I think it is desirable to render the manifests and resource names from the helm release but they would only render if I emitted them from Check (and not from Create/Update etc.). As a result Check is an "online"-only operation at the moment. I am happy to open an issue to revisit this. Perhaps we can mark those fields as computed instead when the cluster is unreachable.

Reply:
Yeah, it seems like we should be able to fall back to showing some minimal info rather than erroring. Two things to check here would be YAML render mode, which generally doesn't require an active cluster, and the case where the cluster hasn't been created yet, but is a parent in the stack via the Provider.